packages/firebase_storage/android/src/main/java/io/flutter/plugins/firebase/storage/FirebaseStoragePlugin.java

```diff
@@ -51,12 +53,50 @@ public void onMethodCall(MethodCall call, final Result result) {
       case "StorageReference#getDownloadUrl":
         getDownloadUrl(call, result);
         break;
+      case "StorageReference#getMetadata":
+        getMetadata(call, result);
+        break;
       default:
         result.notImplemented();
         break;
     }
   }
 
+  private void getMetadata(MethodCall call, final Result result) {
+    String path = call.argument("path");
+    StorageReference ref = firebaseStorage.getReference().child(path);
+    ref.getMetadata()
+        .addOnSuccessListener(
+            new OnSuccessListener<StorageMetadata>() {
+              @Override
+              public void onSuccess(StorageMetadata storageMetadata) {
+                HashMap<String, Object> map = new HashMap();
```

Comment:
`Map<String, Object> map = new HashMap<>();`

It is idiomatic Java to "code against interfaces", where a suitable one exists, like `Map` or `List`. This means all variables and parameters are given the interface type.

You get a compiler warning by leaving out the diamond operator `<>`. Unfortunately, the class-level annotation `@SuppressWarnings("unchecked")` suppresses all the help the compiler might provide. I suggest moving that annotation to the innermost level possible; it can be applied to individual methods or (better) to declarations of local variables.

Maybe you'll find that you don't need it at all.